src/data/calculations/ICalculationsRepository.ts

```diff
@@ -25,22 +26,48 @@ export abstract class ICalculationsRepository {
 
     abstract setOngoingCalculation(ongoing: boolean): Promise<void>;
 
-    abstract storeCalculation(
-        calculationData: CalculationDataType
-    ): Promise<void>;
+    abstract storeCalculation(calculationData: CalculationData): Promise<void>;
 
-    abstract getLastCalculation(): Promise<CalculationDataType | null>;
+    abstract getLastCalculation(): Promise<CalculationData | null>;
 
-    abstract getAllCalculations(): Promise<CalculationDataType[]>;
+    abstract getAllCalculations(): Promise<CalculationData[]>;
 }
 
 export type UserType = "new user" | "returning user";
 
-export type CalculationDataType = {
-    bytes: number;
-    emissions: number;
-    specificEmissions: number;
-    selectedCountries: Map<CountryName, number>;
-    unixTimeMs: number;
-    userType: UserType;
-};
+export class CalculationData {
+    constructor(
+        public bytes: number,
+        public emissions: number,
+        public specificEmissions: number,
+        public selectedCountries: Map<CountryName, number>,
+        public unixTimeMs: number,
+        public userType: UserType
+    ) {}
+
+    public toJSON(): string {
+        return JSON.stringify({
+            bytes: this.bytes,
+            emissions: this.emissions,
+            specificEmissions: this.specificEmissions,
+            selectedCountries: maptoJSON(this.selectedCountries),
+            unixTimeMs: this.unixTimeMs,
+            userType: this.userType,
+        });
+    }
```

Comment:
I know a java dev when I see one

Reply:
how would you do it? 😂

Reply:
I would potentially just use a typeguard if i was writting it myself - but I have strayed far from the ideal OOP patterns